src/text.rs

```diff
@@ -372,7 +377,7 @@ impl TextBox {
                 let start_cursor = Cursor::new(line.line_i, range.start);
                 let end_cursor = Cursor::new(line.line_i, range.end);
                 if let Some((highlight_x, highlight_w)) = line.highlight(start_cursor, end_cursor) {
-                    let x = screen_position.0 + highlight_x;
+                    let x = left + highlight_x;
```

Comment:
`left` is already `screen_position.0` for `Align::Left`, but correctly handles the offset for `Center` and `Right` as well (this is why the underlines were stuck left-aligned)